feat: update partitions docs (#475)
canonical/craft-application: docs/howto/partitions.rst

To add partition support to an application, two basic changes are needed:

#. Enable the feature.

   In your Application subclass, override the following method and invoke the
   :class:`Features <craft_parts.Features>` class:

   .. code-block:: python

     from craft_parts import Features

>      class SnackcraftApplication(Application):

Reviewer: While snarky and fun, it might read more straightforward for the casual reader to make this more explicit

```suggestion
     class ExampleApplication(Application):
```

#. Enable the feature

   Use the :class:`Features <craft_parts.Features>` class to specify that the
   application will use partitions.  This should be called from an overridden
   method in your Application subclass:

   .. code-block:: python

     from craft_parts import Features

     class SnackcraftApplication(Application):

         ...

         @override
         def _enable_craft_parts_features(self) -> None:
             Features(enable_partitions=True)

>    .. NOTE::

Reviewer 2: Since this clears up an understandable point of confusion, we can promote this to a regular paragraph. Also, as a general principle I use callouts judiciously because they fight for the reader's attention.

The text could also be refined:

```rest
    You can only be enable partitions with the :class:`Features
    <craft_parts.Features>` class from craft-parts. In craft-applications
    there's a similarly-named :class:`AppFeatures
    <craft_application.AppFeatures>` class which serves a similar purpose, but
    it can't enable partitions.
```

      but partitions cannot be enabled via :class:`AppFeatures
      <craft_application.AppFeatures>`!

>    .. NOTE::
>       In unit tests, you may need to call this between tests, otherwise the
>       :class:`Features <craft_parts.Features>` global singleton may complain as
>       successive tests try to enable partitions repeatedly.

Reviewer 2: ```suggestion
    .. Tip::
      In unit tests, the :class:`Features <craft_parts.Features>` global
      singleton may raise exceptions when successive tests repeatedly try to
      enable partitions.

      To prevent these excess errors, reset the features at the start of each
      test:
```

      .. code-block:: python

        Features.reset()

#. Define the list of partitions

>    Override your :class:`Application <craft_application.Application>`'s
>    ``_setup_partitions`` method and return the list of the partitions.

Reviewer 2: ```suggestion
   Override the  ``_setup_partitions`` method of your :class:`Application
   <craft_application.Application>` class and return the list of the
   partitions.
```

It's better to avoid adding possessives to compound nouns.

Author: Nice
